src/internal/operations/manifests_test.go

```diff
@@ -0,0 +1,612 @@
+				{
+					Manifest: &snapshot.Manifest{},
+					Reasons: []kopia.Reason{
+						{
+							ResourceOwner: ro,
+							Service:       path.ExchangeService,
+							Category:      path.EmailCategory,
+						},
+					},
+				},
+				{
+					Manifest: &snapshot.Manifest{IncompleteReason: "ir"},
+					Reasons: []kopia.Reason{
+						{
+							ResourceOwner: ro,
+							Service:       path.ExchangeService,
+							Category:      path.EmailCategory,
+						},
+					},
+				},
+			},
+			expect: assert.NoError,
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			err := verifyDistinctBases(test.mans)
+			test.expect(t, err)
+		})
+	}
+}
+
+func (suite *OperationsManifestsUnitSuite) TestProduceManifestsAndMetadata() {
+	const (
+		ro  = "resourceowner"
+		tid = "tenantid"
+		did = "detailsid"
+	)
+
+	makeMan := func(pct path.CategoryType, ir, bid string) *kopia.ManifestEntry {
+		return &kopia.ManifestEntry{
+			Manifest: &snapshot.Manifest{
+				IncompleteReason: ir,
+				Tags:             map[string]string{"tag: " + kopia.TagBackupID: bid},
+			},
+			Reasons: []kopia.Reason{
+				{
+					ResourceOwner: ro,
+					Service:       path.ExchangeService,
+					Category:      pct,
+				},
+			},
+		}
+	}
+
+	table := []struct {
+		name          string
+		mr            mockManifestRestorer
+		gdi           mockGetDetailsIDer
+		reasons       []kopia.Reason
+		getMeta       bool
+		assertErr     assert.ErrorAssertionFunc
+		assertB       assert.BoolAssertionFunc
+		expectDCS     []data.Collection
+		expectNilMans bool
+	}{
+		{
+			name: "don't get metadata, no mans",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "don't get metadata",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{makeMan(path.EmailCategory, "", "")},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "don't get metadata, incomplete manifest",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{makeMan(path.EmailCategory, "ir", "")},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "fetch manifests errors",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mrErr:        assert.AnError,
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.Error,
```

Comment:
I'm not sure what the correct behavior here is, but we may actually want to fallback to a full backup and log the error as a warning? Otherwise I think it will stop the backup from running at all

Reply:
I'm not sure either.  Going to leave it as-is for now, and we can revisit as needed if/when concern arises.

Reply:
make a ticket for it and bring it up in our upcoming discussion (I know it's not technically tech debt, but does need to be addressed)

Reply:
Do we need a ticket just to discuss it?

Reply:
I suppose not as long as we remember to discuss it